src/semeio/workflows/ahm_analysis/ahmanalysis.py

```diff
@@ -98,37 +106,69 @@
 """
 
 
-class AhmAnalysisJob(SemeioScript):
+class AhmAnalysisJob(ErtScript):
     """Define ERT workflow to evaluate change of parameters for eac
     observation during history matching
     """
 
     def run(
         self,
-        target_name="analysis_case",
-        prior_name=None,
-        group_by="data_key",
-        output_dir=None,
-    ):
-        # (SemeioScript wraps this run method)
-
+        workflow_args: list[str],
+        storage: Storage,
+        es_settings: ESSettings,
+        observation_settings: UpdateSettings,
+        random_seed: int,
+        reports_dir: str,
+        ensemble: Ensemble,
+    ) -> Any:
         """Perform analysis of parameters change per obs group
         prior to posterior of ahm"""
-        if output_dir is not None:
-            self._reports_dir = output_dir
 
-        obs_keys = list(self.facade.get_observations().obs_vectors.keys())
-        key_map = _group_observations(self.facade, obs_keys, group_by)
+        if not workflow_args:
+            target_name = None
+            prior_name = None
+            group_by = None
+        else:
+            target_name, prior_name, group_by = workflow_args
```

Comment:
User could provide less than 3 args, so should also check the length and only pick out the ones provided.

Reply:
```python
    target_name = workflow_args[0] if len(workflow_args) > 0 else "analysis_case"
    prior_name = workflow_args[1] if len(workflow_args) > 1 else None
    sort_by = workflow_args[2] if len(workflow_args) > 2 else "data_key"
    ```